color/src/temperature.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2025 the Color Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#![expect(
+    clippy::excessive_precision,
+    reason = "We are sticking with the constants from Tanner Helland."
+)]
+
+use crate::{AlphaColor, ColorSpace, OpaqueColor};
+
+#[cfg(all(not(feature = "std"), not(test)))]
+use crate::floatfuncs::FloatFuncs;
+
+// Implementation notes:
+//
+// * These can't be const yet as `clamp`, `ln`, and `powf` are not `const`.
+
+impl<CS: ColorSpace> AlphaColor<CS> {
+    /// Convert a temperature in Kelvin to a color.
+    ///
+    /// The Kelvin temperature will be clamped to a range of 1000-40,000.
+    pub fn from_kelvin(kelvin: f32) -> Self {
+        let [r, g, b] = CS::from_linear_srgb(from_kelvin(kelvin));
+        Self::new([r, g, b, 1.])
+    }
+}
+
+impl<CS: ColorSpace> OpaqueColor<CS> {
+    /// Convert a temperature in Kelvin to a color.
+    ///
+    /// The Kelvin temperature will be clamped to a range of 1000-40,000.
+    pub fn from_kelvin(kelvin: f32) -> Self {
+        Self::new(CS::from_linear_srgb(from_kelvin(kelvin)))
+    }
+}
+
+/// Convert a Kelvin temperature into a linear RGB color.
+fn from_kelvin(kelvin: f32) -> [f32; 3] {
+    let kelvin = kelvin.clamp(1000., 40000.) / 100.;
+
+    let red = if kelvin < 66. {
+        255.
+    } else {
+        329.698_727_446 * (kelvin - 60.).powf(-0.133_204_759_2)
+    };
+
+    let green = if kelvin <= 66. {
+        99.470_802_586_1 * kelvin.ln() - 161.119_568_166_1
+    } else {
+        288.122_169_528_3 * (kelvin - 60.).powf(-0.075_514_849_2)
+    };
+
+    let blue = if kelvin >= 66. {
```

Comment:
The writing on the post about this says that blues above 6500K are 255, but the math here and in other implementations is checking against 6600K. I think this should be fixed.

I also leave things as they are in the sample code as calculating a value between 0 and 255 and then at the end dividing by 255. This should probably do better.